### PR TITLE
Kowshik tokencreation

### PR DESCRIPTION
**Pull request by kowshik-plivo**
Implemented token creation.

### CODE REVIEW

File: plivo/resources/token.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        nbf = [required(of_type(six.text_type))],
+        exp = [required(of_type(six.text_type))],
+        incoming_allowed = [optional(of_type(bool))],
+        outgoing_allowed = [optional(of_type(bool))],
+        app = [optional(of_type(six.text_type))]
+    )
+
+    def create( self,iss, auth_token, sub,nbf, exp, incoming_allowed, outgoing_allowed,app):
+        if(incoming_allowed == True and sub == None):
+                raise ValidationError('Subaccount is required')
```
Comment (prashantp-plivo) [inline plivo/resources/token.py:20]: Here "sub" does not mean subaccount. Change error message to 'sub is required when incoming allow is true'

File: plivo/resources/token.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        auth_token = [required(of_type(six.text_type))],
+        sub = [optional(of_type(six.text_type))],
+        nbf = [required(of_type(six.text_type))],
+        exp = [required(of_type(six.text_type))],
+        incoming_allowed = [optional(of_type(bool))],
+        outgoing_allowed = [optional(of_type(bool))],
+        app = [optional(of_type(six.text_type))]
+    )
+
+    def create( self,iss, auth_token, sub,nbf, exp, incoming_allowed, outgoing_allowed,app):
```
Comment (prashantp-plivo) [inline plivo/resources/token.py:18]: Add validation check for "iss". It is a mandatory field and must be present

Comment (kowshik-plivo) [reply plivo/resources/token.py:18]: Updates are added

**Review by prashantp-plivo — APPROVED** [review]

File: plivo/resources/token.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from plivo.utils import to_param_dict
+from plivo.utils.validators import *
+
+class Token(PlivoResourceInterface):
+    @validate_args(
+        iss = [required(of_type(six.text_type))],
+        sub = [optional(of_type(six.text_type))],
+        nbf = [optional(of_type(six.text_type))],
+        exp = [optional(of_type(six.text_type))],
+        incoming_allowed = [optional(of_type(bool))],
```
Comment (abhishekGupta-Plivo) [inline plivo/resources/token.py:12]: instead of incoming_allowed it should be [incoming_allow]

File: plivo/resources/token.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        outgoing_allowed = [optional(of_type(bool))],
```
Comment (abhishekGupta-Plivo) [inline plivo/resources/token.py:13]: instead of outgoing_allowed it should be [outgoing_allow]

File: plivo/resources/token.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        app = [optional(of_type(six.text_type))]
+    )
+
+    def create( self, iss, sub = None, nbf = None, exp = None, incoming_allowed = None, outgoing_allowed = None, app = None):
```
Comment (abhishekGupta-Plivo) [inline plivo/resources/token.py:17]: outgoing_allowed and incoming_allowed should be replaced with [incoming_allow and outgoing_allow]

File: plivo/resources/token.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        app = [optional(of_type(six.text_type))]
+    )
+
+    def create( self, iss, sub = None, nbf = None, exp = None, incoming_allowed = None, outgoing_allowed = None, app = None):    
+        if(incoming_allowed == True and sub == None):
+            raise ValidationError('Sub is required')
+        elif(iss == None):
+            raise ValidationError('Iss is required')
+        else:
+            return self.client.request('POST', ('JWT','Token', ),to_param_dict(self.create,locals()), is_voice_request=True)
```
Comment (abhishekGupta-Plivo) [inline plivo/resources/token.py:23]: to_param_dict(self.create,locals() has an issue we need to have a patter in JSON like below for call permissions
  `  "per": {
    "voice": {
      "incoming_allow": true,
      "outgoing_allow": true
    }
  }`

File: plivo/resources/token.py (hunk 3 above)
Comment (abhishekGupta-Plivo) [reply plivo/resources/token.py:12]: incoming_allow and outgoing_allow should accept boolean and String type boolean

**Review by abhishekGupta-Plivo — COMMENTED** [review]
Please follow this guideline for the SDK versioning 
https://docs.google.com/document/d/1a_vv-7mr2xmztXHts5l789f9l9vvZIso9IsENLKsYIE/edit

**Comment by narayana-plivo** [comment]
@abhishekGupta-Plivo coverage is very less, Please add more UT.

**Pull request merged**